src/status_im/ui2/screens/chat/composer/input.cljs

```diff
@@ -168,12 +174,130 @@
+           [rn/text (when (= type :mention) {:style {:color colors/primary-50}})
+            text])
+         (get @input-texts chat-id)))]))
+
+(defn selectable-text-input-manager []
+  (when (exists? (.-NativeModules react-native))
+    (.-RNSelectableTextInputManager ^js (.-NativeModules react-native))))
+
+(defonce rn-selectable-text-input (reagent/adapt-react-class (.requireNativeComponent react-native "RNSelectableTextInput")))
+
+(declare first-level-menu-items second-level-menu-items)
+
+(defn update-input-text [{:keys [text-input chat-id]} text]
+  (on-text-change text chat-id)
+  (.setNativeProps ^js text-input (clj->js {:text text})))
+
+(defn calculate-input-text [{:keys [full-text selection-start selection-end]} content]
+  (let [head (subs full-text 0 selection-start)
+        tail (subs full-text selection-end)]
+    (str head content tail)))
+
+(def first-level-menus {:cut (fn [{:keys [content] :as params}]
+                               (let [new-text (calculate-input-text params "")]
+                                 (react/copy-to-clipboard content)
+                                 (update-input-text params new-text)))
+
+                        :copy-to-clipboard (fn [{:keys [content]}]
+                                             (react/copy-to-clipboard content))
+
+                        :paste (fn [params]
+                                 (let [callback (fn [paste-content]
+                                                  (let [content (string/trim paste-content)
+                                                        new-text (calculate-input-text params content)]
+                                                    (update-input-text params new-text)))]
+                                   (react/get-from-clipboard callback)))
+
+                        :biu (fn [{:keys [first-level text-input-handle menu-items selection-start selection-end]}]
+                               (let [show-menu #(.startActionMode (selectable-text-input-manager) text-input-handle)]
+                                 (reset! first-level false)
+                                 (reset! menu-items second-level-menu-items)
+                                 ;to avoid something discusting like this https://lightrun.com/answers/facebook-react-native-textinput-controlled-selection-broken-on-both-ios-and-android
+                                 ;use native invoke instead! do not use setNativeProps! e.g. (.setNativeProps ^js text-input (clj->js {:selection {:start selection-start :end selection-end}}))
+                                 (.setSelection (selectable-text-input-manager) text-input-handle selection-start selection-end)
+                                 ;set a delay to wait RNSelectableTextInputViewManager#_menuItems get updated, need figure out a better way
+                                 (js/setTimeout show-menu 200)))})
+
+(def first-level-menu-items (map i18n/label (keys first-level-menus)))
```

Comment:
Instead of calling keys to iterate over map keys, You can do
`(map (fn [[k _]] (i18n/label k)) first-level-menus)`